yb-voyager/cmd/config.go

```diff
@@ -0,0 +1,474 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+var allowedGlobalConfigKeys = map[string]bool{
+	"export-dir":            true,
+	"log-level":             true,
+	"send-diagnostics":      true,
+	"run-guardrails-checks": true,
+}
+
+var allowedSourceConfigKeys = map[string]bool{
```

Comment:
nit: `mapset.NewThreadUnsafeSet` might be useful instead of creating a map

Reply:
Done